tests/integration_tests/test_cli.py

```diff
@@ -300,3 +301,57 @@ def test_preprocess_cli(csv_filename):
         config_filename = os.path.join(tmpdir, "config.yaml")
         dataset_filename = _prepare_data(csv_filename, config_filename)
         _run_ludwig("preprocess", dataset=dataset_filename, preprocessing_config=config_filename)
+
+
+@pytest.mark.distributed
+@pytest.mark.parametrize("backend", ["local", "horovod"])
+@pytest.mark.parametrize("type_of_run", ["train", "experiment"])
+def test_reproducible_cli_runs(backend, type_of_run, csv_filename):
+    """Test for reproducible training using `ludwig experiment --dataset`."""
+    with tempfile.TemporaryDirectory() as tmpdir:
+        config_filename = os.path.join(tmpdir, "config.yaml")
+        dataset_filename = _prepare_data(csv_filename, config_filename)
+
+        if backend == "local":
+            command_to_run = _run_ludwig
+        else:
+            command_to_run = _run_ludwig_horovod
+
+        # run first model
+        command_to_run(
+            type_of_run,
+            dataset=dataset_filename,
+            config=config_filename,
+            output_directory=tmpdir,
+            skip_save_processed_input="",  # skip saving preprocessed inputs for reproducibility
+            experiment_name="reproducible",
+            model_name="run1",
+            random_seed="1919",
```

Comment:
Consider parameterizing this random seed to make sure that we see the reproducibility for at least one other random seed value.

Reply:
@justinxzhao 
incorporated comments:
* added second random seed
* add test that a different seed generate different results